dj/construction/build.py

```diff
@@ -339,37 +339,65 @@ def _build_select_ast(
     _build_tables_on_select(session, select, tables, build_criteria)
 
 
-def add_filters_dimensions_limit_to_query_ast(
+def add_filters_dimensions_orderby_limit_to_query_ast(
     query: ast.Query,
     dialect: Optional[str] = None,  # pylint: disable=unused-argument
     filters: Optional[List[str]] = None,
     dimensions: Optional[List[str]] = None,
+    orderbys: Optional[List[str]] = None,
     limit: Optional[int] = None,
 ):
     """
     Add filters and dimensions to a query ast
     """
-    projection_addition = []
+    projection_addition = set()
+
+    dimension_tables = set()
+    if dimensions:
+        for agg in dimensions:
+            temp_select = parse(
+                f"select * group by {agg}",
+            ).select
+            query.select.group_by += temp_select.group_by  # type:ignore
+            projection_addition |= set(temp_select.find_all(ast.Column))
+            dimension_tables |= set(temp_select.find_all(ast.Table))
+
     if filters:
         filter_asts = (  # pylint: disable=consider-using-ternary
             query.select.where and [query.select.where] or []
         )
 
         for filter_ in filters:
+            # use parse to get the asts from the strings we got
             temp_select = parse(f"select * where {filter_}").select
             filter_asts.append(
-                # use parse to get the asts from the strings we got
                 temp_select.where,  # type:ignore
             )
+            dimension_tables |= set(temp_select.find_all(ast.Table))
+            projection_addition |= set(temp_select.find_all(ast.Column))
         query.select.where = ast.BinaryOp.And(*filter_asts)
 
-    if dimensions:
-        for agg in dimensions:
-            temp_select = parse(
-                f"select * group by {agg}",
-            ).select
-            query.select.group_by += temp_select.group_by  # type:ignore
-            projection_addition += list(temp_select.find_all(ast.Column))
+    if not query.organization:
+        query.organization = ast.Organization([])
+
+    orderby_tables = set()
+    if orderbys:
+        for order in orderbys:
+            temp_query = parse(
+                f"select * order by {order}",
+            )
+            orderby_tables |= set(temp_query.find_all(ast.Table))
+            projection_addition |= set(temp_query.find_all(ast.Column))
+            query.organization.order += temp_query.organization.order  # type:ignore
+
+    if diff := orderby_tables - dimension_tables:
+        raise DJInvalidInputException(
+            "Order by columns can only be from "
+            "dimensions used in filters or dimensions arguments"
```

Comment:
What is `dimensions arguments` ?

Reply:
tried to clarify in the message since there are api parameters `filters` and `dimensions` but of course you are using dimension columns in `filters` as well